File: arrayPreviousLess/array_previous_less.py

from typing import List

def array_previous_less(items: List[int]) -> List[int]:
	
	aggregate = []

	for i, x in enumerate(items):   
		inferiors = [x for x in items[:i] if x < items[i]]
		aggregate.append(
			-1 if len(inferiors) == 0 
			else inferiors[-1] )
	
	return aggregate


tests = [
	[3, 5, 2, 4, 5],
	[2, 4, 7, 3, 1, 3]
]

for t in tests: 
	print(f'{t} -> {array_previous_less(t)}')

File: arrayPreviousLess/arrayPreviousLess.ts

const arrayPreviousLess = (items: number[]): number[] => {
	return items.map( (item, i) => {

		if ( i === 0 )
		{
			return -1;
		}
		
		const inferiors = items.slice(0, i).filter(inf => inf < item);
		
		return inferiors.length === 0 
			? -1 
			: inferiors.pop();
	});	
}


const tests: number[][] = [
	[3, 5, 2, 4, 5],
	[2, 4, 7, 3, 1, 3]
];

console.time('arrayPreviousLess x2');

tests.forEach(test => {
	console.log(`${test} -> ${arrayPreviousLess(test)}`);
});

console.timeEnd('arrayPreviousLess x2');